### ISSUE TITLE
Add users API endpoint

### ISSUE THREAD

**Issue by jonfroehlich** [ProjectSidewalk/SidewalkWebpage#2890]
To better enable analysis of end users and performance, @misaugstad and I discussed adding a new users API endpoint.

Currently, to do things like https://github.com/ProjectSidewalk/sidewalk-quality-analysis, we need to get a database dump. Also relates to: https://github.com/ProjectSidewalk/SidewalkWebpage/issues/2889

I think the API endpoint would return a list of all users in the system with:

```
{
  user_id : 12345,  // user_id but not real username, like a hash
  total_label_cnt : 12345,
  total_validations_given_cnt : 12345,
  total_validations_received_cnt : 12345,
  overall_validation_accuracy : 76.5%
  labels{
    surface_problems{
      label_cnt : 1245,
      validated_correct : 123,
      validated_incorrect : 123,
      validated_unsure : 123
    }
   obstacles {
     ...
    }
  ...
  }
}

**Comment by misaugstad** [comment]
@jonfroehlich here are a couple other columns that we just have in the `user_stat` table. Which of these do we want to include?

* `meters_audited`: I'm assuming yes
* `labels_per_meter` I'm also assuming yes?
* `high_quality` (whether or not their data is marked as high quality in the database)
* `high_quality_manual` whether they have been manually marked as high/low quality in the database by me

I figure yes to the first two, but not sure about the latter two?

**Comment by misaugstad** [comment]
Which users should this API include? Just registered/researcher/admin, or turkers and anon users as well? Do we want to restrict to just users that have placed a label or performed a validation; the API response won't be too muddied if we exclude anon users, but if we have them, then a minimum contribution might be helpful?

I don't think I see a reason why we wouldn't include anon users and turkers? Amsterdam folks won't need this, but we could theoretically use it in the future. Which makes me think that we require one label (after the tutorial) or one validation to include a user?

**Comment by misaugstad** [comment]
Also just to record it here, I'm planning to do `own_labels_validated` (I'm flexible on the naming) over `total_validations_received_cnt`. I think it's more helpful to have the number of the user's labels that were validated as opposed to the number of validations received.

**Comment by jonfroehlich** [comment]
Great discussion and ideas.

> - meters_audited: I'm assuming yes
> - labels_per_meter I'm also assuming yes?
> - high_quality (whether or not their data is marked as high quality in the database
> - high_quality_manual whether they have been manually marked as high/low quality in the database by me

Yes to all of these.

With regards to which users, I was thinking everyone but maybe not anon? I agree that if we include anon, we need one label after the tutorial or one validation to include them.

I don't understand the distinction between `own_labels_validated` and `total_validations_received_cnt`. Aren't they synonymous?

Note that I'm trying to address the problem with field naming that we have on the admin site right now where it's hard to immediately grok what we mean by the validation fields (confusing!).
![image](https://user-images.githubusercontent.com/1621749/173675561-81446283-9fa4-4c93-b4bc-b5ad7771a0d0.png)

**Comment by jonfroehlich** [comment]
For validation stuff, maybe something like:

```
{
  user_id : 12345,  // user_id but not real username, like a hash
  total_label_cnt : 12345,
  total_labels_not_yet_validated, 12345,
  overall_label_accuracy_from_validations: 75.6%
  total_validations_given_cnt : 12345,
  total_agree_validations_given_cnt: 12345,
  total_disagree_validations_given_cnt: 12345,
  total_unsure_validations_given_cnt: 12345,
  total_validations_received_cnt : 12345,
  total_agree_validations_received_cnt : 12345,
  total_disagree_validations_received_cnt : 12345,
  total_unsure_validations_received_cnt : 12345,
  
}
```

And  `overall_label_accuracy_from_validations` is a computed field like (num of labels with positive agreement score / num of labels validated)

**Comment by misaugstad** [comment]
>I don't understand the distinction between `own_labels_validated` and `total_validations_received_cnt`. Aren't they synonymous?

Sorry, I meant to include the distinction. The former counts the number of _labels_ validated, whereas the latter counts the number of _validations_, of which there could be multiple per label.

>Note that I'm trying to address the problem with field naming that we have on the admin site right now where it's hard to immediately grok what we mean by the validation fields (confusing!).

I love it :grin: The naming convention you gave is really clear, but it implies counting multiple validations per label, which I don't think is as useful of a metric.

More to come!

**Comment by jonfroehlich** [comment]
Gotcha. Let me try again:

```
{
  user_id : 12345,  // user_id but not real username, like a hash
  total_label_cnt : 12345,
  total_labels_not_yet_validated, 12345,
  overall_label_accuracy_from_validations: 75.6%
  total_validations_given_cnt : 12345,
  agree_validations_given_cnt: 12345,
  disagree_validations_given_cnt: 12345,
  unsure_validations_given_cnt: 12345,
  total_validations_received_cnt : 12345,
  total_labels_with_validations : 12345,
  num_labels_with_positive_validations : 12345, // count of user's own labels where agree cnt > disagree cnt
  num_labels_with_negative_validations : 12345, // count of user's own labels where disagree cnt < agree cnt
  num_labels_with_unsure_validations : 12345, // count of user's own labels where agree cnt == disagree cnt or unsure cnt > max(agree, disagree)... something like that?
}
```

**Comment by misaugstad** [comment]
I find all of the "total", "overall" and "cnt" words make it hard for me to get what I'm looking at. Here's a rewrite, let me know if you think I've removed important detail.

```
{
    user_id: 12345,
    labels: 12345,
    label_accuracy: 75.6%,
    validated_labels: 12345,
    validations_received: 12345,
    labels_validated_correct: 12345,
    labels_validated_incorrect: 12345,
    labels_not_validated: 12345,
    validations_given: 12345,
    dissenting_validations_given: 12345,
    agree_validations_given: 12345,
    disagree_validations_given: 12345,
    notsure_validations_given: 12345,
    labels {
        surface_problem {
            labels: 1245,
            validated_correct: 123,
            validated_incorrect: 123,
            validated_notsure: 123
        }
       obstacles {
           ...
        }
}
```

One note: I'm just kind of combining labels that haven't been validated with labels that have 'notsure' validations with labels that have agree == disagree validations. I don't see differentiating between them to be useful here, and just adds complexity to something that is clearly already confusing enough.

**Comment by jonfroehlich** [comment]
Looks good! And I like this crisper language!
